python-avd/pyavd/_eos_designs/structured_config/overlay/route_maps.py

```diff
@@ -19,104 +19,57 @@ class RouteMapsMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def route_maps(self: AvdStructuredConfigOverlayProtocol) -> list | None:
-        """Return structured config for route_maps."""
+    @structured_config_contributor
+    def route_maps(self: AvdStructuredConfigOverlayProtocol) -> None:
+        """Set the structured config for route_maps."""
         if self.shared_utils.overlay_cvx:
-            return None
-
-        route_maps = []
+            return
 
         if self.shared_utils.overlay_routing_protocol == "ebgp":
             if self.inputs.evpn_prevent_readvertise_to_server:
                 remote_asns = natural_sort({rs_dict.get("bgp_as") for rs_dict in self._evpn_route_servers.values()})
                 for remote_asn in remote_asns:
-                    route_map_name = f"RM-EVPN-FILTER-AS{remote_asn}"
-                    route_maps.append(
-                        {
-                            "name": route_map_name,
-                            "sequence_numbers": [
-                                {
-                                    "sequence": 10,
-                                    "type": "deny",
-                                    "match": [f"as {remote_asn}"],
-                                },
-                                {
-                                    "sequence": 20,
-                                    "type": "permit",
-                                },
-                            ],
-                        },
-                    )
+                    route_maps_item = self.structured_config.RouteMapsItem(name=f"RM-EVPN-FILTER-AS{remote_asn}")
+                    sequence_numbers_item = route_maps_item.SequenceNumbersItem(sequence=10, type="deny")
+                    sequence_numbers_item.match.append(f"as {remote_asn}")
+                    route_maps_item.sequence_numbers.append(sequence_numbers_item)
+                    sequence_numbers_item = route_maps_item.SequenceNumbersItem(sequence=20, type="permit")
+                    route_maps_item.sequence_numbers.append(sequence_numbers_item)
```

Comment:
Let's create the object with EosCliConfigGen as done everywhere 
```suggestion
                    route_maps_item = EosCliConfigGen.RouteMapsItem(name=f"RM-EVPN-FILTER-AS{remote_asn}")
                    sequence_numbers_item = EosCliConfigGen.RouteMapsItem.SequenceNumbersItem(sequence=10, type="deny")
                    sequence_numbers_item.match.append(f"as {remote_asn}")
                    route_maps_item.sequence_numbers.append(sequence_numbers_item)
                    sequence_numbers_item = EosCliConfigGen.RouteMapsItem.SequenceNumbersItem(sequence=20, type="permit")
                    route_maps_item.sequence_numbers.append(sequence_numbers_item)
                    self.structured_config.route_maps.append(route_maps_item)
```

Reply:
Fixed!